inc/contract.class.php

```diff
@@ -1267,73 +1265,72 @@ static function cronContract($task = null) {
          }
 
          // Get contrats with periodicity alerts
-         $query_periodicity = "SELECT `glpi_contracts`.*
-                               FROM `glpi_contracts`
-                               WHERE `glpi_contracts`.`alert` & ".pow(2, Alert::PERIODICITY)." > '0'
-                                     AND `glpi_contracts`.`entities_id` = '".$entity."' ";
+         $valPow = pow(2, Alert::PERIODICITY);
+         $query_periodicity = ['FROM' => 'glpi_contracts',
+             'WHERE' => ['alert' => ['&', $valPow],
+                 'entities_id' => $entity,
+                 'is_deleted' => 0
+             ]
+         ];
 
          // Foreach ones :
          foreach ($DB->request($query_periodicity) as $data) {
-            $entity    = $data['entities_id'];
-            // Compute end date + 12 month : do not send alerts after
-            $end_alert = date('Y-m-d',
-                              strtotime($data['begin_date']." +".($data['duration']+12)." month"));
-            if (!empty($data['begin_date'])
-                && $data['periodicity']
-                && ($end_alert > date('Y-m-d'))) {
+            $entity = $data['entities_id'];
+
+            // For contracts with begin date and periodicity
+            if (!empty($data['begin_date']) && $data['periodicity']) {
                $todo = ['periodicity' => Alert::PERIODICITY];
-               if ($data['alert']&pow(2, Alert::NOTICE)) {
-                  $todo['periodicitynotice'] = Alert::NOTICE;
+               if ($data['alert'] & pow(2, Alert::NOTICE)) {
+                  $todo = ['periodicitynotice' => Alert::NOTICE];
```

Comment:
Are you sure for this one? This will replace the whole `$todo`.

Reply:
Nice catch ! Thanks :)